server/iframe.go

```diff
@@ -67,38 +78,64 @@ func (a *API) iFrame(w http.ResponseWriter, r *http.Request) {
 		a.p.API.LogWarn("Unable to serve the iFrame", "error", err.Error())
 	}
 }
-
 func (a *API) iframeNotificationPreview(w http.ResponseWriter, r *http.Request) {
 	userID := r.Header.Get("Mattermost-User-ID")
 	if userID == "" {
 		http.Error(w, "user not authenticated", http.StatusUnauthorized)
 		return
 	}
-
 	postID := r.URL.Query().Get("post_id")
 	if postID == "" {
 		http.Error(w, "post_id is required", http.StatusBadRequest)
 		return
 	}
-
 	post, err := a.p.API.GetPost(postID)
 	if err != nil {
 		http.Error(w, "failed to get post", http.StatusInternalServerError)
 		return
 	}
```

Comment:
Where do we enforce permission checks?

Reply:
I didn't want to hijack more PRs, since we are waiting for prodsec review: #16